sdk/servicebus/azure-servicebus/azure/servicebus/_base_handler.py

```diff
@@ -158,6 +174,41 @@ def __init__(
         self._auth_uri = None
         self._properties = create_properties(self._config.user_agent)
 
+
+    def _convert_connection_string_to_kwargs(self, conn_str, **kwargs):
+        # type: (str, Type, Any) -> Dict[str, Any]
+        host, policy, key, entity_in_conn_str, token, token_expiry = _parse_conn_str(conn_str)
+        queue_name = kwargs.get("queue_name")
+        topic_name = kwargs.get("topic_name")
+        if not (queue_name or topic_name or entity_in_conn_str):
+            raise ValueError("Entity name is missing. Please specify `queue_name` or `topic_name`"
+                             " or use a connection string including the entity information.")
+    
+        if queue_name and topic_name:
+            raise ValueError("`queue_name` and `topic_name` can not be specified simultaneously.")
+    
+        entity_in_kwargs = queue_name or topic_name
+        if entity_in_conn_str and entity_in_kwargs and (entity_in_conn_str != entity_in_kwargs):
+            raise ServiceBusAuthenticationError(
+                "Entity names do not match, the entity name in connection string is {};"
+                " the entity name in parameter is {}.".format(entity_in_conn_str, entity_in_kwargs)
+            )
+    
+        kwargs["fully_qualified_namespace"] = host
+        kwargs["entity_name"] = entity_in_conn_str or entity_in_kwargs
+        # This has to be defined seperately to support sync vs async credentials.
+        kwargs["credential"] = self._create_credential_from_connection_string_parameters(token,
+                                                                                         token_expiry,
+                                                                                         policy,
+                                                                                         key)
+        return kwargs
+
+    def _create_credential_from_connection_string_parameters(self, token, token_expiry, policy, key):
+        if token and token_expiry:
+            return ServiceBusSASTokenCredential(token, token_expiry)
+        elif policy and key:
+            return ServiceBusSharedKeyCredential(policy, key)
```

Comment:
is there a priority here?: if signature / policy + key are both provided, we would use token and ignore policy + key.

Should it be the reverse? -- if policy + key are provided, then we should first use policy + key.

Reply:
Discussed this with .net and JS folks; going to take inspiration from .net's defensive wisdom to disallow if both are provided, since you may then not have clear intent for which security context you'd want to be in, and I'd rather throw that back to the user.

very good callout++